### PR TITLE
surf-1383-hx-fence

### PR DESCRIPTION
**Pull request by cathysiller**
JIRA: surf-1383

### LGTM's
- [x] Dev LGTM
- [x] Zoom LGTM

### CODE REVIEW

**Comment by HelixUI** [comment]
Deploy preview for *helix-ui* processing.

Built with commit acaae2faa3541ce798edaa8d305a88ce957e3902

https://app.netlify.com/sites/helix-ui/deploys/5bac04fb3813f046a403b4b8

File: src/helix-ui/elements/HXDropFenceElement.html [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<div id="hxDropFence">
+  <hx-file-icon type="paperclip"></hx-file-icon>
```
Comment (CITguy) [inline src/helix-ui/elements/HXDropFenceElement.html]: Design Q: Will this always be a paperclip? Would it ever be configurable?

File: src/helix-ui/elements/HXDropFenceElement.less [hunk 2]
```diff
@@ -0,0 +1,10 @@
+@import "./HXElement";
+
+#hxDropFence {
+  align-items: center;
+  display: flex;
+  flex-direction: column;
+  height: 100%;
+  justify-content: center;
+  width: 100%;
+}
```
Comment (CITguy) [inline src/helix-ui/elements/HXDropFenceElement.less]:
```less
#hxDropFence {
  > div {
    max-width: 60ch;
  }
}
```

If we expect a drop fence to _always_ have content, we'll also need to add a `margin-top: 0.5rem;`.

File: docs/components/files/test-drop-fences.html [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    </nav>
+  </section>
+
+  <section>
+    <h2>Drop Fence (Empty)</h2>
+    <hx-drop-fence>
+    </hx-drop-fence>
+  </section>
+
+  <section>
```
Comment (CITguy) [inline docs/components/files/test-drop-fences.html]: We'll probably want some test cases for a fence with text node children.

```html
<hx-drop-fence>I'm plain text</hx-drop-fence>
```

File: src/helix-ui/styles/elements/hx-drop-fence.less [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+hx-drop-fence {
+  #Box.md();
+  border: 1px dashed @gray-600;
+  display: block;
+  font-size: 0.875rem;
+  text-align: center;
+
+  > *:first-child {
+    margin-top: 0.5rem;
```
Comment (CITguy) [inline src/helix-ui/styles/elements/hx-drop-fence.less]: This doesn't work for text-node children `<hx-drop-fence>I'm text</hx-drop-fence>`

File: docs/components/files/test-drop-fences.html (hunk 3 above)
Comment (cathysiller) [reply docs/components/files/test-drop-fences.html]: added -- look below in file

File: src/helix-ui/elements/HXDropFenceElement.less (hunk 2 above)
Comment (cathysiller) [reply src/helix-ui/elements/HXDropFenceElement.less]: added, look at css file -- converted to rem

File: docs/elements/hx-drop-fence/index.html [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+{% block content %}
+  <section>
+    <p>
+      The custom <code>{{page.title}}</code> element provides a visual area
+      of the boundaries of where a user can drag a file to.
+    </p>
+  
+    <dl class="hxBox-md metadata hxList">
+      <div>
+        <dt>Permitted Parents</dt>
```
[the remark this answers is not in the document]
Comment (cathysiller) [reply docs/elements/hx-drop-fence/index.html]: added, pls look at file

**Pull request merged**